spec/models/ems_event_spec.rb

```diff
@@ -519,4 +519,25 @@
       end
     end
   end
+
+  context 'Physical Storage Events' do
+    let(:ems)   { FactoryBot.create(:ems_autosde) }
+    let(:physical_storage) { FactoryBot.create(:physical_storage_autosde, :ems_ref => "ems1", :ext_management_system => ems) }
+    before do
+      @event_hash = {
+        :event_type               => "autosde_critical_alert",
+        :ems_ref                  => "1",
+        :physical_storage_ems_ref => physical_storage.ems_ref,
+        :ems_id                   => ems.id,
+        :message                  => "description"
+      }
+    end
+
+    it "test process_physical_storage_in_event!" do
+      EmsEvent.process_physical_storage_in_event!(@event_hash)
```

Comment:
We want use the "interface" method rather than internal methods (actually surprised these aren't private)
```suggestion
      EmsEvent.add!(ems.id, event_hash)
```